hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/ECBlockOutputStream.java

```diff
@@ -115,20 +115,43 @@ ContainerCommandResponseProto> executePutBlock(boolean close,
     }
 
     BlockData checksumBlockData = null;
+    boolean foundStripeChecksum = false;
+    BlockID blockID = null;
     //Reverse Traversal as all parity will have checksumBytes
     for (int i = blockData.length - 1; i >= 0; i--) {
       BlockData bd = blockData[i];
       if (bd == null) {
         continue;
       }
+      if (blockID == null) {
+        // store the BlockID for logging
+        blockID = bd.getBlockID();
+      }
       List<ChunkInfo> chunks = bd.getChunks();
-      if (chunks != null && chunks.size() > 0 && chunks.get(0)
-          .hasStripeChecksum()) {
-        checksumBlockData = bd;
-        break;
+      if (chunks != null && chunks.size() > 0) {
```

Comment:
The original logic had:

```
if (chunks != null && chunks.size() > 0 && chunks.get(0)
          .hasStripeChecksum()) {
```

So that `checksumBlockData` is only set if `hasStripeChecksum())` returns true. With the change you have made, `checksumBlockData` will be set if `hasChecksumData()`, but it might not have stripeChecksum in it.

Then at line 155 it will enter the IF block and at line 174 I am not sure what will happen if stripeCheckSum is missing.

All the IF statement at 155 does is copy in the stripeChecksum if it exists. So if it does not exist, there is no point in going into that IF at all, as we will just be copying the original chunkChecksum out and back in again.

Following on from this - we only set `checksumBlockData` if there is a stripeChecksum, then you can also remove the `foundStripeChecksum` boolean as `checksumBlockData != null` means the same thing.

Reply:
Ah, so what you're saying is that there's no point in going inside the if block at line 155 if `stripeChecksum` isn't found because that code is only setting `stripeChecksum`. So overall, the only change we need here is to log instead of throw at line 184. Right?

Reply:
Yea, that's it.